h2o-algos/src/main/java/hex/gram/Gram.java

```diff
@@ -758,6 +758,39 @@ public void mul(double [] x, double [] res){
     }
   }
 
+  /*
+  This method will not allocate the extra memory and hence is considered for lowMemory systems.
+  However, need to consider case when you have categoricals!  Make them part of the matrix
+  in the multiplication process.
+   */
```

Comment:
I misunderstood the older method thinking that the memory allocation is essential to certain applications.  You are right, only one is needed.  I replaced the older method as I noticed there are some numerical issues with the older one.